contrib/http_util/src/main/java/io/opencensus/contrib/http/HttpHandler.java

```diff
@@ -0,0 +1,213 @@
+   * @since 0.13
+   */
+  public HttpExtractor<Q, P> getExtractor() {
+    return this.extractor;
+  }
+
+  /**
+   * Returns a span name determined by the given parameters.
+   *
+   * @param request the request entity.
+   * @param method the qualified name of the method which handles the request sending procedure,
+   *     which will be used when constructing the span name. Although user can obtain such
+   *     information from the request itself, this parameter provides a more convenient way to
+   *     specify it directly.
+   * @return a span name determined by the given parameters.
+   * @since 0.13
+   */
+  public abstract String getSpanName(Q request, String method);
+
+  /**
+   * Generate a sequential id for sent {@link MessageEvent}.
+   *
+   * @return next id for sent {@code MessageEvent}.
+   * @since 0.13
+   */
+  public long nextSentMessageEventId() {
+    return sentMessageEventId.getAndIncrement();
+  }
+
+  /**
+   * Generate a sequential id for received {@link MessageEvent}.
+   *
+   * @return next id for received {@code MessageEvent}.
+   * @since 0.13
+   */
+  public long nextRecvMessageEventId() {
+    return recvMessageEventId.getAndIncrement();
+  }
+
+  /**
+   * Records a {@link MessageEvent} with given parameters.
+   *
+   * @param span the span which this {@code MessageEvent} will be added to.
+   * @param id the id of the event.
+   * @param type the {@code MessageEvent.Type} of the event.
+   * @param uncompressedMessageSize size of the message before compressed (optional).
+   * @param compressedMessageSize size of the message after compressed (optional).
+   * @since 0.13
+   */
+  public void recordMessageEvent(
+      Span span, long id, Type type, long uncompressedMessageSize, long compressedMessageSize) {
+    checkNotNull(span, "span");
+    checkNotNull(type, "type");
+    MessageEvent messageEvent =
+        MessageEvent.builder(type, id)
+            .setUncompressedMessageSize(uncompressedMessageSize)
+            .setCompressedMessageSize(compressedMessageSize)
+            .build();
+    span.addMessageEvent(messageEvent);
+  }
+
+  /**
+   * Parse OpenCensus Status from HTTP response.
+   *
+   * <p>This method provides a default implementation to map HTTP status code to Open Census Status.
+   * The mapping is defined in <a
+   * href="https://github.com/googleapis/googleapis/blob/master/google/rpc/code.proto">Google API
+   * canonical error code</a>.
+   *
+   * <p>Users can override this method to define their own parsing behaviors.
+   *
+   * @param response the HTTP response entity. It could be null if error is present.
+   * @param error the error occured during response transmition. It could be null if response is
+   *     present.
+   * @return the corresponding OpenCensus {@code Status}.
+   * @since 0.13
+   */
+  public Status parseResponseStatus(@Nullable P response, @Nullable Throwable error) {
```

Comment:
I think we should make this an independent util function which accepts an int status and maybe an error.